src/outputs/outputs.hpp

```diff
@@ -177,44 +179,24 @@ class VTKOutput : public OutputType {
   void WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) override;
 };
 
-//----------------------------------------------------------------------------------------
-//! \class RestartOutput
-//  \brief derived OutputType class for restart dumps
-
-class RestartOutput : public OutputType {
- public:
-  explicit RestartOutput(const OutputParameters &oparams) : OutputType(oparams) {
-#ifndef HDF5OUTPUT
-    std::stringstream msg;
-    msg << "### FATAL ERROR in Restart (Outputs) constructor" << std::endl
-        << "Executable not configured for HDF5 outputs, but HDF5 file format "
-        << "is requested in output block '" << output_params.block_name << "'"
-        << std::endl;
-    PARTHENON_FAIL(msg);
-#endif
-  }
-  void WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) override;
-};
-
 #ifdef HDF5OUTPUT
 //----------------------------------------------------------------------------------------
 //! \class PHDF5Output
-//  \brief derived OutputType class for Athena HDF5 files
+//  \brief derived OutputType class for Athena HDF5 files or restart dumps
 
 class PHDF5Output : public OutputType {
  public:
   // Function declarations
-  explicit PHDF5Output(const OutputParameters &oparams) : OutputType(oparams) {}
+  PHDF5Output(const OutputParameters &oparams, bool restart)
```

Comment:
Make bool const reference

Reply:
POD (int, bool, double, etc.) should be passed by value, not reference. Passing by reference can be slightly less performant since it requires two memory reads (first reading the pointer, then following it) and a pointer is actually a bigger word than a bool/int.

Reply:
Ah, thanks for the correction, I would just make it const then.